api/app/Http/Controllers/BaseController.php

```diff
@@ -1,69 +1,66 @@
 <?php namespace App\Http\Controllers;
 
-use App;
+use App\Http\Transformers\CollectionTransformerInterface;
+use App\Http\Transformers\ItemTransformerInterface;
+use App\Repositories\BaseRepository;
+use App\Services\Validator;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Laravel\Lumen\Routing\Controller;
-use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+use Symfony\Component\HttpFoundation\Response;
 use Illuminate\Http\Request;
-use Illuminate\Database\Eloquent\Model;
-use Illuminate\Database\Eloquent\Collection;
 
 abstract class BaseController extends Controller
 {
-    public static $model;
-
     /**
      * Model Repository.
      *
-     * @var App\Repositories\BaseRepository
+     * @var BaseRepository
      */
     protected $repository;
 
     /**
      * Validation service.
      *
-     * @var App\Services\Validation\TestValidator
+     * @var Validator
      */
     protected $validator;
 
-    /**
-     * Transformer to use for responses.
-     *
-     * @var string
-     */
-    protected $transformer = 'App\Http\Transformers\BaseTransformer';
 
     /**
      * Get all entities.
      *
+     * @param CollectionTransformerInterface $transformer
      * @return Response
      */
-    public function getAll()
+    public function getAll(CollectionTransformerInterface $transformer)
     {
-        return $this->collection($this->repository->all());
+        return response($transformer->transformCollection($this->repository->all()));
```

Comment:
We should discuss this, I don't see a need to duplicate the injection of the transformer. I makes much more sense for the developer to return an `item` or a `collection` of items